full-service/src/json_rpc/v2/e2e_tests/transaction/build_submit/build_then_submit.rs

```diff
@@ -749,126 +755,102 @@ mod e2e_transaction {
+                let body = json!({
+                    "jsonrpc": "2.0",
+                    "id": 1,
+                    "method": "submit_transaction",
+                    "params": {
+                        "tx_proposal": tx_proposal,
+                        "account_id": account.id,
+                    }
+                });
+                let res = dispatch(&client, body, &logger);
+                let result = res.get("result").unwrap();
+                let submitted_tx_log_id = result
+                    .get("transaction_log")
+                    .unwrap()
+                    .get("id")
+                    .unwrap()
+                    .as_str()
+                    .unwrap();
+                assert_eq!(tx_log_id, submitted_tx_log_id);
+            });
+
+        let tx_proposal = TxProposal::try_from(&tx_log_id_and_proposals[0].1).unwrap();
+
+        add_block_with_tx(&mut ledger_db, tx_proposal.tx, &mut rng);
         manually_sync_account(
             &ledger_db,
             &db_ctx.get_db_instance(logger.clone()),
-            &AccountID(account_id.to_string()),
+            &AccountID(account.id),
             &logger,
         );
 
-        // Get the transaction_id and verify it contains what we expect
-        let body = json!({
-            "jsonrpc": "2.0",
-            "id": 1,
-            "method": "get_transaction_log",
-            "params": {
-                "transaction_log_id": submitted_transaction_id,
-            }
-        });
-        let res = dispatch(&client, body, &logger);
-        let result = res.get("result").unwrap();
-        let transaction_log: TransactionLog =
-            serde_json::from_value(result.get("transaction_log").unwrap().clone()).unwrap();
-
-        assert_eq!(transaction_log.status, TxStatus::Succeeded.to_string());
-        assert_eq!(transaction_log.id, submitted_transaction_id);
-
-        // Get the transaction_id and verify it contains what we expect
-        let body = json!({
-            "jsonrpc": "2.0",
-            "id": 1,
-            "method": "get_transaction_log",
-            "params": {
-                "transaction_log_id": txlog_id_b,
-            }
-        });
-        let res = dispatch(&client, body, &logger);
-        let result = res.get("result").unwrap();
-        let transaction_log: TransactionLog =
-            serde_json::from_value(result.get("transaction_log").unwrap().clone()).unwrap();
-        assert_eq!(transaction_log.id, txlog_id_b);
-        assert_eq!(transaction_log.status, TxStatus::Built.to_string());
+        // The first transaction log will have succeeded, because it's outputs
+        // show up on the ledger.
+        // The second transaction log will still be pending, because it's outputs
+        // don't exist in the ledger. This second transaction log will
+        // eventually fail once the tombstone block is reached.
+        let expected_statuses = [TxStatus::Succeeded, TxStatus::Pending];
```

Comment:
If one reverts the code changes in `full-service/src/db/transaction_log.rs` and `full-service/src/service/sync.rs` this test will fail with both status coming back as `Succeeded`